File: archive/legacy-services/shared/models/category.model.ts
// Category Model - نموذج التصنيفات الأساسي

export interface Category {
  id: string;
  nameAr: string;
  nameEn: string;
  icon?: string;
  image?: string;
  productCount: number;
  subcategories?: Category[];
  parentId?: string;
  level: number;
  sortOrder: number;
  isActive: boolean;
  isFeatured: boolean;
  searchKeywords: string[];
  createdAt: Date;
  updatedAt?: Date;
}

export interface CategoryFilters {
  isActive?: boolean;
  isFeatured?: boolean;
  level?: number;
  parentId?: string;
}

export interface CreateCategoryRequest {
  nameAr: string;
  nameEn: string;
  icon?: string;
  image?: string;
  parentId?: string;
  sortOrder?: number;
  isActive?: boolean;
  isFeatured?: boolean;
  searchKeywords?: string[];
}

export interface UpdateCategoryRequest extends Partial<CreateCategoryRequest> {
  id: string;
}

// Category Tree - شجرة التصنيفات الكاملة
export interface CategoryTree {
  categories: Category[];
  totalCount: number;
  maxLevel: number;
}

// Default Categories - التصنيفات الافتراضية (مثل إيباي)
export const DEFAULT_CATEGORIES: Category[] = [
  {
    id: 'electronics',
    nameAr: 'إلكترونيات',
    nameEn: 'Electronics',
    icon: '📱',
    productCount: 0,
    level: 1,
    sortOrder: 1,
    isActive: true,
    isFeatured: true,
    searchKeywords: ['موبايل', 'لابتوب', 'كمبيوتر', 'تلفزيون'],
    createdAt: new Date(),
  },
  {
    id: 'fashion',
    nameAr: 'موضة',
    nameEn: 'Fashion',
    icon: '👕',
    productCount: 0,
    level: 1,
    sortOrder: 2,
    isActive: true,
    isFeatured: true,
    searchKeywords: ['ملابس', 'أحذية', 'إكسسوارات', 'عطور'],
    createdAt: new Date(),
  },
  {
    id: 'home',
    nameAr: 'المنزل',
    nameEn: 'Home',
    icon: '🏠',
    productCount: 0,
    level: 1,
    sortOrder: 3,
    isActive: true,
    isFeatured: true,
    searchKeywords: ['أثاث', 'أجهزة', 'ديكور', 'مطبخ'],
    createdAt: new Date(),
  },
  {
    id: 'vehicles',
    nameAr: 'مركبات',
    nameEn: 'Vehicles',
    icon: '🚗',
    productCount: 0,
    level: 1,
    sortOrder: 4,
    isActive: true,
    isFeatured: true,
    searchKeywords: ['سيارات', 'دراجات', 'قطع غيار', 'إطارات'],
    createdAt: new Date(),
  },
  {
    id: 'sports',
    nameAr: 'رياضة',
    nameEn: 'Sports',
    icon: '⚽',
    productCount: 0,
    level: 1,
    sortOrder: 5,
    isActive: true,
    isFeatured: true,
    searchKeywords: ['ملابس رياضية', 'معدات', 'أحذية رياضية', 'لياقة'],
    createdAt: new Date(),
  },
];

File: mobile-app-flutter/lib/features/categories/models/category_model.dart
import 'package:equatable/equatable.dart';

class Category extends Equatable {
  final String id;
  final String nameAr;
  final String nameEn;
  final String? icon;
  final String? image;
  final int productCount;
  final List<Category>? subcategories;
  final String? parentId;
  final int level;
  final int sortOrder;
  final bool isActive;
  final bool isFeatured;
  final List<String> searchKeywords;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Category({
    required this.id,
    required this.nameAr,
    required this.nameEn,
    this.icon,
    this.image,
    this.productCount = 0,
    this.subcategories,
    this.parentId,
    this.level = 1,
    this.sortOrder = 0,
    this.isActive = true,
    this.isFeatured = false,
    this.searchKeywords = const [],
    required this.createdAt,
    this.updatedAt,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      nameAr: json['nameAr'] ?? json['name'],
      nameEn: json['nameEn'] ?? json['name'],
      icon: json['icon'],
      image: json['image'],
      productCount: json['productCount'] ?? 0,
      subcategories: json['subcategories'] != null
          ? (json['subcategories'] as List).map((e) => Category.fromJson(e)).toList()
          : null,
      parentId: json['parentId'],
      level: json['level'] ?? 1,
      sortOrder: json['sortOrder'] ?? 0,
      isActive: json['isActive'] ?? true,
      isFeatured: json['isFeatured'] ?? false,
      searchKeywords: List<String>.from(json['searchKeywords'] ?? []),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'nameAr': nameAr,
        'nameEn': nameEn,
        'icon': icon,
        'image': image,
        'productCount': productCount,
        'subcategories': subcategories?.map((e) => e.toJson()).toList(),
        'parentId': parentId,
        'level': level,
        'sortOrder': sortOrder,
        'isActive': isActive,
        'isFeatured': isFeatured,
        'searchKeywords': searchKeywords,
        'createdAt': createdAt.toIso8601String(),
        'updatedAt': updatedAt?.toIso8601String(),
      };

  @override
  List<Object?> get props => [id, nameAr, nameEn];

  // Helper methods
  bool get hasSubcategories => subcategories != null && subcategories!.isNotEmpty;
  bool get isRootCategory => parentId == null;
  String get displayName => nameAr; // Use Arabic name by default
}